source/common/access_log/access_log_formatter.cc

```diff
@@ -23,6 +23,9 @@ static const std::string UnspecifiedValueString = "-";
 
 namespace {
 
+// Matches newline pattern in a StartTimeFormatter format string.
+const std::regex& getNewlinePattern(){CONSTRUCT_ON_FIRST_USE(std::regex, "%[-_0^#]*[1-9]*n")};
```

Comment:
I think it should be `?` rather than `*`, since there can only be one flag according to the man page. Also, is it possible to have negative field widths? printf supports these, would want to verify we don't need to deal with them as well..

Reply:
Oops, didn't catch this. It seems like both "%^^8n" and "%^8n" and "%8n" will produce a new line after 8 spaces, and that you may put as many flag in as you want.